### PR TITLE
Removed findelement by nsPredicate string

### PR DESCRIPTION
**Pull request by SrinivasanTarget**
## Change list

1)Removed findelement by nsPredicate string
2)findelement and findelements methods of RemoteWebdriver is now public
 
## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details

1)Removed findelement by nsPredicate string
          @rafael-chavez Apologies for removing this now.https://github.com/appium/java-client/pull/352 When XCUIT mode will work then we will commit these changes again with same authorship(rafael-chavez).
2)findelement and findelements methods of RemoteWebdriver is now public
           Above methods are made public so that consumers of JSONWP in other projects can utilise this easily.

As we discussed earlier, IOSDriver will be split into IOSAutomationDriver and IOSXCUITDriver in upcoming PR's soon.

@TikhomirovSergey @rafael-chavez Please review.

### CODE REVIEW

File: src/main/java/io/appium/java_client/DefaultGenericMobileDriver.java [hunk 1]
```diff
@@ -59,6 +59,14 @@ public DefaultGenericMobileDriver(CommandExecutor executor, Capabilities desired
         return (T) super.findElement(by);
     }
 
+    @Override public T findElement(String by, String using) {
+        return (T) super.findElement(by, using);
+    }
+
+    @Override public List findElements(String by, String using) {
```
Comment (TikhomirovSergey) [inline src/main/java/io/appium/java_client/DefaultGenericMobileDriver.java]: This method shpuld be overriden at AppiumDriver as well. Just take a look at
https://github.com/appium/java-client/blob/master/src/main/java/io/appium/java_client/AppiumDriver.java#L223
for instance.

Comment (SrinivasanTarget) [inline src/main/java/io/appium/java_client/DefaultGenericMobileDriver.java]: Sure.Added here now: https://github.com/appium/java-client/pull/394/files#diff-c60f84c0cc90f37a9a4513eeebd4896dR231

File: src/main/java/io/appium/java_client/AppiumDriver.java [hunk 2]
```diff
@@ -228,6 +228,14 @@ private static CommandInfo deleteC(String url) {
         return super.findElementsById(id);
     }
 
+    @Override public T findElement(String by, String using) {
```
Comment (TikhomirovSergey) [inline src/main/java/io/appium/java_client/AppiumDriver.java]: It is not necessary to add _@Override public T findElement(String by, String using)_ 
but it is neccessary to 
 _@Override public List<T> findElements(String by, String using)_ (please pay attention)

**Comment by rafael-chavez** [comment]
@SrinivasanTarget, what do you mean by 'XCUIT mode will work'? Currently appium-xcuitest-driver does work with when used in conjunction with the nspredicate changes in the java-client.

**Comment by SrinivasanTarget** [comment]
@rafael-chavez We accept nspredicate work but other UIAutomation methods won't work now as per current status/architecure of java-client repo.So as you are aware iOSDriver is going to be split into XCUITDriver and IOSAutomationDriver soon.Once this work is done, we will get this back to repo again with same authorship.

https://github.com/appium/java-client/pull/352 will be in again once driver split up is done.
Thanks for your support.

File: src/main/java/io/appium/java_client/AppiumDriver.java (hunk 2 above)
Comment (SrinivasanTarget) [inline src/main/java/io/appium/java_client/AppiumDriver.java]: I'm on it.Will close this and styling issues ASAP.

**Comment by rafael-chavez** [comment]
Great. Looking forward to the Split up driver work. Thanks @SrinivasanTarget

**Comment by SrinivasanTarget** [comment]
Thanks for your support @rafael-chavez . Split up will happen soon in upcoming PR's.

**Comment by SrinivasanTarget** [comment]
@TikhomirovSergey Have fixed the remarks

File: src/main/java/io/appium/java_client/AppiumDriver.java [hunk 3]
```diff
@@ -224,6 +224,10 @@ private static CommandInfo deleteC(String url) {
         return super.findElements(by);
     }
 
+    @Override public List findElements(String by, String using) {
```
Comment (TikhomirovSergey) [inline src/main/java/io/appium/java_client/AppiumDriver.java]: @SrinivasanTarget 
You forgot to change this string to 

```java
@Override public List<T> findElements(String by, String using)
```

**Comment by TikhomirovSergey** [comment]
Everything is ok. But there is the remark above.

**Pull request merged**